src/entry_writer.rs

```diff
@@ -19,8 +19,25 @@ impl<'a, W: Write> EntryWriter<'a, W> {
     }
 
     fn write_entry(writer: &mut W, entry: &Entry) -> io::Result<()> {
-        let serialized = serde_json::to_string(entry).unwrap();
-        writeln!(writer, "{}", serialized)?;
+        let raw =
+            bincode::serialize(&entry).map_err(|e| Error::new(ErrorKind::Other, e.to_string()))?;
+        // TODO quote new lines...
+
+        trace!("write_entry raw = {:?}", raw);
+
+        let mut escaped = Vec::with_capacity(raw.len());
+        for mut c in raw {
+            if c == b'\\' || c == b'\n' {
+                trace!("escaping '{}' to {}", c, c + b'\\');
+                escaped.push(b'\\');
+                c += b'\\';
+            }
+            escaped.push(c);
+        }
+        trace!("write_entry escaped = {:?}", escaped);
+
+        writer.write(&escaped[..])?;
```

Comment:
I still don't understand why the binary format couldn't be tuples of `(raw.len(),raw)` with no escaping.

Reply:
good idea, I'll try that approach

Reply:
most of the trouble with the len->bytes approach is my feebleness with rust...  constructing an iterator from something that's calling read() read()...